OpenRA.Mods.Common/Lint/CheckSequences.cs

```diff
@@ -98,6 +101,30 @@ static void Run(Action<string> emitError, Action<string> emitWarning, Ruleset ru
 										if (!sequences.Sequences(i).Any(s => s.StartsWith(sequence)))
 											emitWarning($"Actor type `{actorInfo.Value.Name}` trait `{traitName}` field `{field.Name}` defines a prefix `{sequence}` that does not match any sequences on image `{i}`.");
 									}
+									else if (sequenceReference.FactionSuffix)
+									{
+										var hasFallback = sequences.HasSequence(i, sequence);
+										var sequencesWithPrefix = sequences.Sequences(i).Where(s => s.StartsWith(sequence) && s != sequence).ToHashSet();
+										if (sequencesWithPrefix.Count > 0)
+										{
+											foreach (var faction in selectableFactions)
+											{
+												var fullSequence = $"{sequence}.{faction}";
+												if (sequences.HasSequence(i, fullSequence))
+													sequencesWithPrefix.Remove(fullSequence);
+												else if (!hasFallback)
+													emitError($"Actor type `{actorInfo.Value.Name}` trait `{traitName}` field `{field.Name}` references faction specific sequence `{fullSequence}` on image `{i}`, which does not exist.");
+											}
+
+											foreach (var unkownSequence in sequencesWithPrefix)
+											{
+												var suffix = unkownSequence[(sequence.Length + 1)..];
+												emitWarning($"Actor type `{actorInfo.Value.Name}` trait `{traitName}` field `{field.Name}` expects faction specific sequences with prefix `{sequence}` on image `{i}`, but suffix '{suffix}' does not belong to any faction.");
```

Comment:
```suggestion
												emitWarning($"Actor type `{actorInfo.Value.Name}` trait `{traitName}` field `{field.Name}` has invalid faction specific sequence `{unknownSequence}` on image `{i}`, faction '{suffix}' is not a selectable faction.");
```

 I think this wording is clearer. Might also bee good to rename `suffix` to `faction`